core/includes/form.inc

```diff
@@ -3629,17 +3629,21 @@ function form_process_container($element, &$form_state) {
  */
 function theme_container($variables) {
   $element = $variables['element'];
-  // Ensure #attributes is set.
+  // Ensure #attributes and classes within it are set; preserve existing ones.
   $element += array('#attributes' => array());
+  $element['#attributes'] += array('class' => array());
 
   // Special handling for form elements.
   if (isset($element['#array_parents'])) {
-    // Assign an html ID.
+    // Assign an HTML ID.
     if (!array_key_exists('id', $element['#attributes'])) {
       $element['#attributes']['id'] = $element['#id'];
     }
-    // Add the form-wrapper class.
-    if (empty($element['#attributes']['class'])) {
+    // Add the form-wrapper class if not already added, but only for 'container'
+    // elements, otherwise #states does not work properly for 'checkboxes' and
+    // 'radios' elements.
+    // See: https://github.com/backdrop/backdrop-issues/issues/711#issuecomment-84353916
```

Comment:
I would remove the @see here, the explanation is enough :)

Reply:
Done 👍🏼 ...see: [`f398185` (#3689)](https://github.com/backdrop/backdrop/pull/3689/commits/f39818515d0fb73d9e9dfc3dba170318016e44d9)